plugins/techdocs-backend/src/helpers.ts

```diff
@@ -119,14 +119,20 @@ export const checkoutGithubRepository = async (
   const token = process.env.GITHUB_PRIVATE_TOKEN || '';
 
   if (fs.existsSync(repositoryTmpPath)) {
-    const repository = await Repository.open(repositoryTmpPath);
-    const currentBranchName = (await repository.getCurrentBranch()).shorthand();
-    await repository.fetch('origin');
-    await repository.mergeBranches(
-      currentBranchName,
-      `origin/${currentBranchName}`,
-    );
-    return repositoryTmpPath;
+    try {
+      const repository = await Repository.open(repositoryTmpPath);
+      const currentBranchName = (
+        await repository.getCurrentBranch()
+      ).shorthand();
+      await repository.fetch('origin');
+      await repository.mergeBranches(
+        currentBranchName,
+        `origin/${currentBranchName}`,
+      );
+      return repositoryTmpPath;
+    } catch {
+      fs.removeSync(repositoryTmpPath);
```

Comment:
It used to bubble up the exception to the caller, now it just swallows it and keeps running, is that intentional?

Also, do we not generate a new temp path for each run?

Reply:
Nope, we keep the repos as a cache to decrease build times. We should probably check that this is one of the issues we can fix by removing the temp folder and just re-throw the error if it isn't.